auth-agents-common/src/main/java/org/apache/atlas/policytransformer/CachePolicyTransformerImpl.java

```diff
@@ -0,0 +1,476 @@
+
+        List<String> users = (List<String>) atlasPolicy.getAttribute("policyUsers");
+        List<String> groups = (List<String>) atlasPolicy.getAttribute("policyGroups");
+        List<String> roles = (List<String>) atlasPolicy.getAttribute("policyRoles");
+
+        List<RangerPolicyItemAccess> accesses = new ArrayList<>();
+        List<String> actions = (List<String>) atlasPolicy.getAttribute("policyActions");
+
+        actions.forEach(action -> accesses.add(new RangerPolicyItemAccess(action)));
+
+
+        if ("allow".equals(policyType)) {
+            RangerPolicyItem item = new RangerPolicyItem();
+            item.setUsers(users);
+            item.setGroups(groups);
+            item.setRoles(roles);
+            item.setAccesses(accesses);
+
+            rangerPolicy.setPolicyItems(Collections.singletonList(item));
+
+        } else if ("deny".equals(policyType)) {
+            RangerPolicyItem item = new RangerPolicyItem();
+            item.setUsers(users);
+            item.setGroups(groups);
+            item.setRoles(roles);
+            item.setAccesses(accesses);
+
+            rangerPolicy.setDenyPolicyItems(Collections.singletonList(item));
+
+        } else if ("allowExceptions".equals(policyType)) {
+            RangerPolicyItem item = new RangerPolicyItem();
+            item.setUsers(users);
+            item.setGroups(groups);
+            item.setRoles(roles);
+            item.setAccesses(accesses);
+
+            rangerPolicy.setAllowExceptions(Collections.singletonList(item));
+
+        } else if ("denyExceptions".equals(policyType)) {
+            RangerPolicyItem item = new RangerPolicyItem();
+            item.setUsers(users);
+            item.setGroups(groups);
+            item.setRoles(roles);
+            item.setAccesses(accesses);
+
+            rangerPolicy.setDenyExceptions(Collections.singletonList(item));
+
+        } else if ("dataMask".equals(policyType)) {
+
+            rangerPolicy.setPolicyType(RangerPolicy.POLICY_TYPE_DATAMASK);
+
+            RangerDataMaskPolicyItem item = new RangerDataMaskPolicyItem();
+            item.setUsers(users);
+            item.setGroups(groups);
+            item.setRoles(roles);
+            item.setAccesses(accesses);
+
+            String maskType = (String) atlasPolicy.getAttribute(ATTR_POLICY_MASK_TYPE);
+
+            if (StringUtils.isEmpty(maskType)) {
+                LOG.error("MASK type not found");
+                throw new AtlasBaseException("MASK type not found");
+            }
+
+            RangerPolicyItemDataMaskInfo dataMaskInfo  = new RangerPolicyItemDataMaskInfo(maskType, null, null);
+            item.setDataMaskInfo(dataMaskInfo);
+
+            rangerPolicy.setDataMaskPolicyItems(Collections.singletonList(item));
+
+        } else if ("rowFilter".equals(policyType)) {
+            rangerPolicy.setPolicyType(RangerPolicy.POLICY_TYPE_ROWFILTER);
+            //TODO
```

Comment:
anything missing ?

Reply:
rowFilter is not supported via product yet, we can add support this later